---
title: C6255
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 216f9bf7225f1590e27d578dfc10baa76c8374ce
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 04/23/2019
ms.locfileid: "62782083"
---
# <a name="c6255"></a>C6255
Aviso C6255: alloca indicou falha, gerando uma exceção de estouro de pilha. Considere utilizar malloca em vez disso

 Este aviso indica que uma chamada para `_alloca` foi detectada fora do tratamento de exceção de local. `_alloca` deve sempre ser chamado de dentro do intervalo protegido de um manipulador de exceção porque pode lançar uma exceção de estouro de pilha em caso de falha. Se possível, em vez de usar `_alloca`, considere o uso `_malloca` que é uma versão mais segura de `_alloca`.

## <a name="example"></a>Exemplo
 O código a seguir gera este aviso, pois `_alloca` possa gerar exceção:

```
#include <windows.h>

void f( )
{
  void *p = _alloca(10);
  // code ...
}
```

 Para corrigir este aviso, use `_malloca` e adicione o manipulador de exceção, conforme mostrado no código a seguir:

```
#include <windows.h>
#include <malloc.h>

void f( )
{
  void *p;
  int errcode;
  __try
  {
    p = _malloca(10);
    // code...
    _freea(p);
  }
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    errcode = _resetstkoflw();
    // code ...
  }
}
```

## <a name="see-also"></a>Consulte também
 [_malloca](/cpp/c-runtime-library/reference/malloca)